src/main/java/Duke.java

```diff
@@ -1,10 +1,155 @@
+                        e.markAsDone();
+                    }
+                    list.add(e);
+                    break;
+                case("D"):
+                    Deadline d = new Deadline(split[2], split[3]);
+                    if (!split[1].equals("0")) {
+                        d.markAsDone();
+                    }
+                    list.add(d);
+                    break;
+                }
+                line = b.readLine();
+            }
+            b.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static void writeAll() {
+        try {
+            FileWriter w = new FileWriter("./Data/Duke.txt");
+            for (Task task : list) {
+                w.write(task.toWrite());
+            }
+            w.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        String logo = "DUKE\n";
+        System.out.println("Hello from\n" + logo + "What can I do for you?");
+        readAll();
+        Scanner s = new Scanner(System.in);
+        String command = s.nextLine();
+        while(!command.equals("bye")) {
+            switch(command.split(" ")[0]) {
+            case "list":
+                System.out.println("Here are the tasks in your list:");
+                for (int i = 0; i < list.size(); i++) {
+                    int num = i + 1;
+                    if (list.get(i) != null) {
+                        System.out.println(num + "." + list.get(i).toString());
+                    }
+                }
+                command = s.nextLine();
+                break;
+            case "done":
+                int finished = Integer.parseInt(command.split(" ")[1]) - 1;
+                list.get(finished).markAsDone();
+                System.out.println("Nice! I've marked this task as done:\n" + "  " + list.get(finished).toString());
+                command = s.nextLine();
+                break;
+            case "todo":
+                try {
+                    if (command.equals("todo") || command.equals("todo ")) {
+                        throw new DukeException();
+                    }
+                    Todo toAdd = new Todo(command.split(" ", 2)[1]);
+                    list.add(toAdd);
+                    update(toAdd);
+                } catch (DukeException exception) {
+                    System.out.println(exception.getMessage());
+                }
+                command = s.nextLine();
+                break;
+            case "deadline":
+                String[] splitD = command.split(" ", 2)[1].split(" /by ", 2);
```

Comment:
Perhaps `splitD` could be renamed as `splitDeadline`, similar to `splitE` below?